### PR TITLE
Book Detail Screen About and Review

### PR DESCRIPTION
**Pull request by khanhnn-1791**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9955
### Goals :soccer:

Create UI for book detail screen
+ Back icon
+ Navigation Bar Title
+ Book image
+ Title, author label
+ Rating View
+ ""Want to read"" this book Button
+ 2 pager view strip About and Review
+ in About view includes:
 - General infomation about book pages and publisher
 - Author infomation
 - A button to know more about author
 - Similiar, Summary and description label
 - A webview to view all summary
 - Simliar books by author using collectionview
+ Buy and Read Button
+ in Review view includes:
 - A webView
Create a UI action
- Click back button to go to the previous view
- Click Want to read to go to the progress view
- Click Buy and Read Button to redirect to amazon/ goodread website for buying this book
- Scroll to see more summary
Fetch data from API and display to the UI
### Implementation Details :construction:

https://gfycat.com/pleasantunsteadyfirefly-michelle-obama

### CODE REVIEW

**Comment by khanhnn-1791** [comment]
ready

File: ios-i-book-lover/Data/Model/Book/BookDetail.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//  BookLover
+//
+//  Created by nguyen.nam.khanh on 4/7/19.
+//  Copyright © 2019 nguyen.nam.khanh. All rights reserved.
+//
+
+import XMLMapper
+
+final class BookDetail: XMLMappable {
+    var nodeName: String! = ""
```
Comment (tgdong2296) [inline ios-i-book-lover/Data/Model/Book/BookDetail.swift:12]:
```suggestion
    var nodeName = ""
```

File: ios-i-book-lover/Scenes/Home/BookDetail/AboutBookViewController.swift [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+extension AboutBookViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: Cells.weightBookCell, height: Cells.heightBookCell)
+    }
+}
+
+extension AboutBookViewController: IndicatorInfoProvider {
+    func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo {
+        return IndicatorInfo(title: "About")
```
Comment (tgdong2296) [inline ios-i-book-lover/Scenes/Home/BookDetail/AboutBookViewController.swift]: `"About"` cho vào constant

File: ios-i-book-lover/Scenes/Home/BookDetail/ReviewBookViewController.swift [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        webView.do {
+            $0.navigationDelegate = self
+            $0.load(URLRequest(url: url))
+        }
+    }
+}
+
+extension ReviewBookViewController: IndicatorInfoProvider {
+    func indicatorInfo(for pagerTabStripController: PagerTabStripViewController) -> IndicatorInfo {
+        return "Review"
```
Comment (tgdong2296) [inline ios-i-book-lover/Scenes/Home/BookDetail/ReviewBookViewController.swift]: cho vào constant

File: ios-i-book-lover/Extensions/UIView+Extension.swift [hunk 4]
```diff
@@ -47,4 +47,31 @@ extension UIView {
         }
         isUserInteractionEnabled = true
     }
+    
+    func activityStartOverlay() {
+        let overlay = UIView(frame: self.frame).then {
```
Comment (tienpx3010) [inline ios-i-book-lover/Extensions/UIView+Extension.swift]: Bỏ self. Tương tự

File: ios-i-book-lover/Scenes/Home/BookDetail/AboutBookViewController.swift [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                }
+                self.collectionView.reloadData()
+            case .failure(let error):
+                print(error ?? "")
+            }
+        }
+    }
+    
+    @IBAction private func knowMoreAction(_ sender: UIButton) {
+        guard let link = bookDetail.author?.link, let url = URL(string: link) else {
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Home/BookDetail/AboutBookViewController.swift]:
```suggestion
        guard let link = bookDetail.author?.link, let url = URL(string: link) else { return }
```

File: ios-i-book-lover/Scenes/Home/BookDetail/ReviewBookViewController.swift [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        print(error.localizedDescription)
+    }
+    
+    func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        print("start")
+    }
+    
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        print("finish")
```
Comment (tienpx3010) [inline ios-i-book-lover/Scenes/Home/BookDetail/ReviewBookViewController.swift]: Bỏ

**Review by tgdong2296 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**